pyonfx/timestamps.py

```diff
@@ -0,0 +1,681 @@
+        index: int,
+        normalize: bool,
+        rounding_method: RoundingMethod,
+    ) -> Timestamps:
+        """Create timestamps based on the ``video_path`` provided.
+
+        Note:
+            This method requires the ``ffprobe`` program to be available.
+
+        Parameters:
+            video_path (os.PathLike[str]): A Video path.
+            index (int, optional): Stream index of the video.
+            normalize (bool, optional): If True, it will shift the timestamps to make them start from 0. If false, the option does nothing.
+            rounding_method (RoundingMethod, optional): A rounding method. See the comment in Timestamps description about floor vs round.
+        Returns:
+            An Timestamps instance.
+        """
+
+        def get_timestamps(packets) -> Tuple[Fraction, Fraction, List[int]]:
+            timestamps: List[int] = []
+
+            lowest_timestamp: Fraction = None
+            highest_timestamp: Fraction = None
+
+            for packet in packets:
+                timestamp = Fraction(
+                    packet.get("pts_time", packet.get("dts_time"))
+                ) * Fraction(1000)
+                if highest_timestamp is None or highest_timestamp < timestamp:
+                    highest_timestamp = timestamp
+                if lowest_timestamp is None or lowest_timestamp > timestamp:
+                    lowest_timestamp = timestamp
+                timestamps.append(rounding_method(timestamp))
+
+            timestamps.sort()
+            return lowest_timestamp, highest_timestamp, timestamps
+
+        # Verify if ffprobe is installed
+        if shutil.which("ffprobe") is None:
+            raise Exception("ffprobe is not in the environment variable.")
+
+        cmd = [
+            "ffprobe",
+            "-select_streams",
+            f"{index}",
+            "-show_entries",
+            "packet=pts_time,dts_time:stream=codec_type,time_base:format=format_name",
+            f"{video_path}",
+            "-print_format",
+            "json",
+        ]
+        ffprobe_output = subprocess.run(cmd, capture_output=True, text=True)
+        ffprobe_output_dict = json.loads(ffprobe_output.stdout)
+
+        if not ffprobe_output_dict:
+            raise ValueError(f"The file {video_path} is not a video file.")
+
+        if len(ffprobe_output_dict["streams"]) == 0:
+            raise ValueError(f"The index {index} is not in the file {video_path}.")
+
+        if ffprobe_output_dict["streams"][0]["codec_type"] != "video":
+            raise ValueError(
+                f'The index {index} is not a video stream. It is an "{ffprobe_output_dict["streams"][0]["codec_type"]}" stream.'
+            )
+
+        if ffprobe_output_dict["format"]["format_name"] == "matroska,webm":
+            # We only do this check for .mkv file. See the note about mkv in the class documentation
+            time_base = Fraction(ffprobe_output_dict["streams"][0]["time_base"])
```

Comment:
Currently, the ``_from_mkvextract`` method doesn't perform a check for the time_base like ffmpeg because when I wrote this code, it was a information that mkvtoolnix didn't expose.
But, I asked if it could be added [here](https://help.mkvtoolnix.download/t/how-to-get-timestamp-scale-information-without-using-mkvinfo).
It was introduced in version 82.0, which was released on 2 January 2024. We could perform a version check and if the user has a V82 or higher, then we perform the check.

Also, I propose to add a new file named ``mkvtoolnix.py`` and extract the timestamps in that file since ``timestamps.py`` is becoming too big.